packages/integration-sdk-cli/src/generator/entitiesFlow.ts

```diff
@@ -0,0 +1,190 @@
+import { askAgain } from './util';
+import { snakeCase } from 'lodash';
+
+export type Entity = {
+  resourceName: string;
+  _type: string;
+  _class: string[];
+};
+
+async function entitiesFlow(inquirer, vendorName): Promise<Entity[]> {
+  const entities: Entity[] = [];
+  const useEntityFlow = await inquirer.prompt({
+    type: 'confirm',
+    name: 'value',
+    message: 'Do you want to declare entities?',
+  });
+  if (!useEntityFlow.value) return entities;
+
+  let again = false;
+  do {
+    const entity = await createEntityPrompt(inquirer, vendorName);
+    entities.push(entity);
+    again = await askAgain(inquirer);
+  } while (again);
+  return entities;
+}
+
+async function createEntityPrompt(inquirer, vendorName) {
+  const { resourceName } = await inquirer.prompt({
+    type: 'input',
+    name: 'resourceName',
+    message:
+      'What is the name of the entity? ex. User, EC2 Instance, Access Key',
+  });
+
+  const { _class } = await inquirer.prompt({
+    type: 'checkbox-plus',
+    name: '_class',
+    message: 'Which classes should the entity have?',
+    searchable: true,
+    pageSize: 10,
+    validate(input) {
+      if (input?.length > 0) {
+        return true;
+      } else {
+        // TODO: It would be nice to use an error message here
+        // but the library for checkbox-plus doesn't render it
+        // correctly
+        return false;
+      }
+    },
+    source: function (ans, input) {
+      return new Promise((resolve) => {
+        if (!input) {
+          resolve(entityClassesChoices);
+          return;
+        } else {
+          const choices = entityClassesChoices.filter(({ name }) =>
+            name.toLowerCase().includes(input.toLowerCase()),
+          );
+          resolve(choices);
+        }
+      });
+    },
+  });
+
+  const { _type } = await inquirer.prompt({
+    Type: 'input',
+    name: '_type',
+    message: 'What should the _type be?',
+    default: `${snakeCase(vendorName)}_${snakeCase(resourceName)}`,
+    validate(input) {
+      const valid = /^[a-z0-9_]+$/.test(input);
```

Comment:
Are we doing `_type` validation in `integration-sdk-runtime`? Is this the same method?

Reply:
Seems like we don't.